Assets/Scripts/Gameplay/GameplayObjects/ServerDropItemOnFaint.cs

```diff
@@ -0,0 +1,43 @@
+using Unity.Netcode;
```

Comment:
Since players can carry only while the action is running, couldn't this logic be part of the action while it's running?

Reply:
Ideally, yes. However, at the moment, there isn't a way to stop an action of the same type to not be added to the action queue, besides playing with the `m_LastUsedTimestamps` which turns out to be quite messy. I'd be setting an unreachable resuse timeout on the action, and check for input inside this action class to drop the item, and it's still not guaranteed to fire after the input checking inside `ClientInputSender`. It would rework a bit of the underlying Action system, so I can definitely add it. But as of writing this, Actions are almost always just added to the action queue, meaning that this class has to do both the pickup and drop logic anytime it's fired.

Reply:
Update:
* Pickup and drop actions have been separated. This allowed for simplification of action flows and for the addition of drop icon.
* This class has since been removed since this is tracked inside of `PickupAction.cs`. Life state is checked on update to determine when to end action.